include/deal.II/sundials/ida_interface.h

```diff
@@ -0,0 +1,476 @@
+     * @param ic_alpha Initial condition Newton parameter
+     * @param ic_max_iter Initial condition Newton max iterations
+     * @param max_non_linear_iterations Maximum number of nonlinear iterations
+     * @param verbose Show output of time steps
+     * @param use_local_tolerances Use local tolerances
+     */
+    IDAInterface(const MPI_Comm mpi_comm = MPI_COMM_WORLD,
+                 const double &initial_time = 0.0,
+                 const double &final_time = 1.0,
+                 const double &initial_step_size = 1e-4,
+                 const double &min_step_size = 1e-6,
+                 const double &abs_tol = 1e-6,
+                 const double &rel_tol = 1e-5,
+                 const unsigned int &max_order = 5,
+                 const double &output_period = .1,
+                 const bool &ignore_algebraic_terms_for_errors = true,
+                 const std::string &ic_type = "use_y_diff",
+                 const std::string &reset_type = "use_y_dot",
+                 const double &ic_alpha = .33,
+                 const unsigned int &ic_max_iter = 5,
+                 const unsigned int &max_non_linear_iterations = 10,
+                 const bool &verbose = true,
+                 const bool &use_local_tolerances = false);
+
+    /**
+     * House cleaning.
+     */
+    ~IDAInterface();
+
+    /**
+     * Add all internal parameters to the given ParameterHandler object. When
+     * the paramaters are parsed from a file, the internal parameters are automatically
+     * updated.
+     *
+     * The following parameters are declared:
+     *
+     * @code
+     * set Absolute error tolerance                      = 0.000001
+     * set Final time                                    = 1.000000
+     * set Ignore algebraic terms for error computations = true
+     * set Initial condition Newton max iterations       = 5
+     * set Initial condition Newton parameter            = 0.330000
+     * set Initial condition type                        = use_y_diff
+     * set Initial condition type after restart          = use_y_dot
+     * set Initial step size                             = 0.000100
+     * set Initial time                                  = 0.000000
+     * set Maximum number of nonlinear iterations        = 10
+     * set Maximum order of BDF                          = 5
+     * set Min step size                                 = 0.000001
+     * set Relative error tolerance                      = 0.000010
+     * set Show output of time steps                     = true
+     * set Time units between each output                = 0.05
+     * set Use local tolerances                          = false
+     * @endcode
+     *
+     * These are one-to-one with the options you can pass at construction time.
+     *
+     * Those options are set as default value in the ParameterHandler object
+     * `prm`.
+     */
+    virtual void add_parameters(ParameterHandler &prm);
```

Comment:
maybe `set_parameters()`? You're basically just *overwriting* existing ones, not adding them.

Reply:
Actually I'm calling `prm.add_parameter` for each parameter, which first declare the parameter, and then sets it. I  don't know if `add_parameter` was a good choice in the first place... but here is just using the same name (with a plural).

Reply:
Oh, I misread (or didn't read at all) the documentation. I thought you were setting the parameters of the current object. But you were really setting the parameters of the current class *in the parameter handler object*.

That's the equivalent of the `declare_parameters()` function we describe in the documentation of `ParameterHandler`.

Reply:
Not exactly... the `declare_parameters` does not attach to each parameter an action, while this one does.

Reply:
Yes, you're right. It's not equivalent. But maybe reference that function in the documentation and say that it combines the action of `declare_parameters()` and `read_parameters()`.

Reply:
In other words, this function also **sets** the parameters whenever `prm` updates its entries.